src/integration_tests/mission_survey.cpp

```diff
@@ -0,0 +1,351 @@
+}
+
+void receive_upload_mission_result(Mission::Result result)
+{
+    EXPECT_EQ(result, Mission::Result::SUCCESS);
+
+    if (result == Mission::Result::SUCCESS) {
+        _mission_state = MissionState::UPLOADING_DONE;
+    } else {
+        LogErr() << "Error: mission send result: " << Mission::result_str(result);
+        _mission_state = MissionState::ERROR;
+    }
+}
+
+void receive_start_mission_result(Mission::Result result)
+{
+    EXPECT_EQ(result, Mission::Result::SUCCESS);
+
+    if (result == Mission::Result::SUCCESS) {
+        _mission_state = MissionState::STARTING_DONE;
+    } else {
+        LogErr() << "Error: mission start result: " << Mission::result_str(result);
+        _mission_state = MissionState::ERROR;
+    }
+}
+
+void receive_mission_progress(int current, int total)
+{
+    LogInfo() << "Mission status update: " << current << " / " << total;
+
+    if (current > 0 && current == total) {
+        _mission_state = MissionState::MISSION_DONE;
+    }
+}
+
+void receive_arm_result(Action::Result result)
+{
+    EXPECT_EQ(result, Action::Result::SUCCESS);
+
+    if (result == Action::Result::SUCCESS) {
+        _mission_state = MissionState::ARMING_DONE;
+    } else {
+        LogErr() << "Error: arming result: " << Action::result_str(result);
+        _mission_state = MissionState::ERROR;
+    }
+}
+
+void receive_return_to_launch_result(Action::Result result)
+{
+    EXPECT_EQ(result, Action::Result::SUCCESS);
+
+    if (result == Action::Result::SUCCESS) {
+    } else {
+        LogErr() << "Error: return to land result: " << Action::result_str(result);
+        _mission_state = MissionState::ERROR;
+    }
+}
+
+void receive_disarm_result(Action::Result result)
+{
+    EXPECT_EQ(result, Action::Result::SUCCESS);
+
+    if (result == Action::Result::SUCCESS) {
+    } else {
+        LogErr() << "Error: disarming result: " << Action::result_str(result);
+    }
+
+    _mission_state = MissionState::DONE;
+}
+
+std::shared_ptr<MissionItem> add_waypoint(double latitude_deg,
+                                          double longitude_deg,
+                                          float relative_altitude_m,
+                                          float speed_m_s,
+                                          bool is_fly_through,
+                                          float gimbal_pitch_deg,
+                                          float gimbal_yaw_deg,
+                                          bool take_photo)
+{
+    std::shared_ptr<MissionItem> new_item(new MissionItem());
+    new_item->set_position(latitude_deg, longitude_deg);
+    new_item->set_relative_altitude(relative_altitude_m);
+    new_item->set_speed(speed_m_s);
+    new_item->set_fly_through(is_fly_through);
+    new_item->set_gimbal_pitch_and_yaw(gimbal_pitch_deg, gimbal_yaw_deg);
+    if (take_photo) {
+        new_item->set_camera_action(MissionItem::CameraAction::TAKE_PHOTO);
+    }
+    return new_item;
```

Comment:
I believe this is a rebase/merge mistake. This test has been removed.

Reply:
Arf, I hope I did not do too much of that :(. Good catch!

Reply:
Wait, which test do you mean? The whole file?

Reply:
Yep, it's gone.

Reply:
Ok, I removed it!